### ISSUE TITLE
upgrade to Java 17 (LTS)

### ISSUE THREAD

**Issue by jrhizor** [airbytehq/airbyte#6096]
It'd be nice to move to Java 17. Records! Speed improvements! Guarantees of long term support!

Unfortunately Gradle doesn't support Java 17, and when they do it will likely be on a 7+ version. Therefore this is blocked by https://github.com/airbytehq/airbyte/issues/4006

**Comment by welly87** [comment]
Hi All, i somehow working on porting airbyte to graviton2/arm so i need to change into the latest version of Java and Gradle. please check this one

https://github.com/weltam/airbyte/tree/gv2/graviton2

i hope the commits is also helping

https://github.com/weltam/airbyte/commits/gv2

**Comment by welly87** [comment]
another article that can help for porting to 17

https://www.infoq.com/articles/why-how-upgrade-java17/

also a couple of samples from 

https://github.com/johanjanssen/JavaUpgrades

**Comment by jrhizor** [comment]
Cloud + OSS have been upgraded to Java 17. https://github.com/airbytehq/airbyte/issues/8708 is a followup ticket for the connectors team to make sure Java connectors are updated and released on Java 17 images.
